test_conformance/api/test_api_consistency.cpp

```diff
@@ -1151,3 +1151,148 @@ int test_consistency_3d_image_writes(cl_device_id deviceID, cl_context context,
 
     return TEST_PASS;
 }
+
+int test_consistency_requirements_fp64(cl_device_id deviceID,
+                                       cl_context context,
+                                       cl_command_queue queue, int num_elements)
+{
+    cl_int error = CL_SUCCESS;
+    cl_device_fp_config value = 0;
+
+    if (is_extension_available(deviceID, "cl_khr_fp64"))
+    {
+        const Version version = get_device_cl_version(deviceID);
+
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_DOUBLE_FP_CONFIG,
+                                sizeof(value), &value, nullptr);
+        test_error(error, "Unable to get device CL_DEVICE_DOUBLE_FP_CONFIG");
+        test_assert_error(
+            value > 0, "CL_DEVICE_DOUBLE_FP_CONFIG must return nonzero value");
+        if (version < Version(2, 0))
+        {
+            test_assert_error(
+                value
+                    & (CL_FP_FMA | CL_FP_ROUND_TO_NEAREST | CL_FP_ROUND_TO_ZERO
+                       | CL_FP_ROUND_TO_INF | CL_FP_INF_NAN | CL_FP_DENORM),
+                "Reported double fp config doesn't meet minimum set "
+                "for OpenCL 1.0, OpenCL 1.1, OpenCL 1.2 devices");
+        }
+        else
+        {
+            test_assert_error(
+                value
+                    & (CL_FP_FMA | CL_FP_ROUND_TO_NEAREST | CL_FP_INF_NAN
+                       | CL_FP_DENORM),
+                "Reported double fp config doesn't meet minimum set "
+                "for OpenCL 2.0 or newer devices");
+        }
+
+        error =
+            clGetDeviceInfo(deviceID, CL_DEVICE_PREFERRED_VECTOR_WIDTH_DOUBLE,
+                            sizeof(value), &value, nullptr);
+        test_error(
+            error,
+            "Unable to get device CL_DEVICE_PREFERRED_VECTOR_WIDTH_DOUBLE");
+        test_assert_error(value > 0,
+                          "CL_DEVICE_PREFERRED_VECTOR_WIDTH_DOUBLE must return "
+                          "nonzero value");
+
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_NATIVE_VECTOR_WIDTH_DOUBLE,
+                                sizeof(value), &value, nullptr);
+        test_error(error,
+                   "Unable to get device CL_DEVICE_NATIVE_VECTOR_WIDTH_DOUBLE");
+        test_assert_error(
+            value > 0,
+            "CL_DEVICE_NATIVE_VECTOR_WIDTH_DOUBLE must return nonzero value");
```

Comment:
So we go in with eyes open: this change requires `CL_DEVICE_NATIVE_VECTOR_WIDTH_DOUBLE` to be non-zero when fp64 is supported.  This is true even if there is no "native" double precision support, say when `CL_DEVICE_DOUBLE_FP_CONFIG` includes `CL_FP_SOFT_FLOAT`.

No action required, but if we did want to "fix" this, some options might be:

1. Remove the check entirely - the spec doesn't have any requirements for a value for `CL_DEVICE_NATIVE_VECTOR_WIDTH_DOUBLE` when fp64 is supported.  Note, there is a requirement with fp64 is NOT supported, so this check should not change.
2. Have the check only when `CL_DEVICE_DOUBLE_FP_CONFIG` does not include `CL_FP_SOFT_FLOAT`, since there should be some "native" support in this case.

Same comments also apply to fp16.

Reply:
Discussed in the October 1st teleconference.  We are going to leave this as-is for now.  We will revisit in the future, if needed, if a vendor wants to return zero for `CL_DEVICE_NATIVE_VECTOR_WIDTH_DOUBLE` and still support fp64 for some reason.